management-cluster-access/README.md

```diff
@@ -0,0 +1,95 @@
+
+## References
+
+- [PKI epic](https://github.com/giantswarm/giantswarm/issues/15981)
+- [Issue regarding client cert creation for CAPO ](https://github.com/giantswarm/giantswarm/issues/21740)
+- [Story for backup IDP in dex](https://github.com/giantswarm/roadmap/issues/603)
+- [Proposal to use Azure AD for GS staff](https://github.com/giantswarm/giantswarm/issues/21627)
+- 
+
+## Possible solutions
+
+### Client certificates
+
+In the vintage product, client certificates can be created from outside the cluster using `vault` for signing.
+At the moment this is not the case for the CAPI product.
+
+- This method of authentication is completely separate from SSO and does not have shared dependencies.
+- As of now, client certificate creation is harder to monitor and control than SSO.
+- Client certificates should be short lived and therefore would need to be recreated by users frequently.
+- A long living client-certificate granting admin access is a security risk since permissions can not be revoked.
+- We do not want this to be the primary authentication method for human users. (That should remain SSO)
+- If we want to support this, we need to decide whether we want to use vault or something else.
+
+### Storing kubeconfig in lastpass
+
+In the CAPI product, a valid kubeconfig is stored in lastpass as emergency fallback. This is part of the bootstrap process.
+
+- This method of authentication is completely separate from SSO and does not have shared dependencies.
+- It is not integrated in our tooling and we need to pull the kubeconfig using lastpass cli.
+- We should revisit how this works. What type of authorization is used here? Is access revokable? Is this rotated?
+- Since it is already integrated, it makes sense to keep it for emergencies while there is no alternative.
+
+### Introduce a second identity provider for SSO
+
+Dex supports a wide array of identity providers. Customers are already using many of them. At the present moment, giant swarm staff can only login using github.
+However, we could add more providers.
+
+- We favor SSO when it comes to identity management for human engineers.
+- Could be introduced for workload cluster access as well.
+- Using e.g. azure AD would make sense since customers are using it.
+- We might be able to simplify account automation when using another identity provider than github.
+- We would still have a dependency on dex and this method will not completely remove the single point of failure.
+- Adding another provider might mean individual setup for each existing configuration. (e.g. in github individual apps are needed) We should avoid this if possible since it already does not scale.
+- While dex allows an array of connectors, we currently use a fixed naming for our single `giantswarm` connector. We need to revisit what relies on this convention to ensure the second connector is equal.
+- Regardless of other means auf authentication being implemented, adding another dex idp would be beneficial.
```

Comment:
My 2 cts here is that using GitHub as identity provider isn't ideal, as it needs a separate OAuth app for every (management and workload) cluster we use Dex with.

So I would definitely favour a more flexible provider like Azure AD, where we can have a single application configured with multiple callback URLs. That should be our default IdP. Our second (backup) IdP should be as flexible as the primary one.

Regarding connector IDs: I think Dex requires unique connector identifiers, so one could be `giantswarm1` and the second could be `giantswarm2`. Of course this has an effect on our RBAC bindings, as the connector IDs have to be present in the group names. So instead of having one group `giantswarm:my-admin-group-name` become admins, we need two groups to be bound: `giantswarm1:my-admin-group-name-from-idp1`, `giantswarm2:my-admin-group-name-from-idp2`.

Reply:
I agree about github. It's a hassle especially when we look at rolling out oidc access for gs staff to workload clusters, it just does not scale.
At the same time, we are using it now and it works. We would likely have to do something like adding another provider (e.g. azure) first, then eventually adding another one and then removing github/no longer supporting it in new installations. 

As for the connector naming, we have to get it right from the beginning. Using `giantswarm1` etc could work or perhaps something more specific like `giantswarm-ad`. We should make sure that connectors and related rbac configuration can be easily removed, replaced and added. 

When it comes to selecting the connector, I personally like the idea of the selection menu. It is in line with how SSO login is generally presented and we wouldn't have to know/manage flag values if something changes (Looking at SSO on WCs here, too)
Shouldn't the extra click be fine for the most part? We do not need to log in again too often since our context can be refreshed.